repos/system_upgrade/el9toel10/actors/sssd/sssdupdate/tests/unit_test_sssdupdate.py

```diff
@@ -0,0 +1,273 @@
+import os
+import random
+
+import pytest
+
+from leapp.libraries.actor import sssdupdate
+from leapp.models import KnownHostsProxyConfig
+
+
+class MockedFile:
+    """
+    Mocks a file to avoid writing to the filesystem.
+
+    This is the minimum mocking required for this test.
+    Most of the file functions are not implemented.
+    Only those required by this test.
+    """
+    def __init__(self, name: str, fs):
+        self.name = name
+        self.fs = fs
+        self.fd = -1
+        self.mode = 0o644
+        self.contents = ''
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+        return False
+
+    def open(self, fd: int, mode: str):
+        self.fd = fd
+        if mode in ['w', 'x']:
+            self.contents = ''
+
+    def fileno(self):
+        return self.fd
+
+    def write(self, text: str) -> int:
+        self.contents += text
+        return len(text)
+
+    def __iter__(self):
+        return iter(self.contents.splitlines(keepends=True))
+
+    def close(self):
+        self.fs._close(self.fd)
+        self.fd = -1
+
+
+class MockedStatResults:
+    def __init__(self, mode):
+        self.st_mode = mode
+
+
+class MockedFileSystem:
+    """
+    Mocks a filesystem.
+
+    Simulates the minimum services needed by the tests in this file.
+    """
+    def __init__(self):
+        self._by_name = {}
+        self._by_fd = []
+
+    def _get_by_fd(self, fd: int) -> MockedFile:
+        try:
+            return self._by_fd[fd]
+        except IndexError:
+            raise OSError('Bad file descriptor')
+
+    def _get_by_name(self, name: str) -> MockedFile:
+        try:
+            return self._by_name[name]
+        except KeyError:
+            raise FileNotFoundError(name)
```

Comment:
nitpick; seems as dead code as I do not see any unit-tests testing the raised exceptions